_transcripts/006-rust-1.39.0.md

```diff
@@ -0,0 +1,783 @@
+too.
+
+__Ben__: By-move bindings, you mentioned?
+
+__Jon__: Yeah. Okay, so this is something that is really annoying if you run
+into it. But you probably don't know about this problem if you haven't run into
+it. Specifically, the issue is if you match on a type, and in the pattern, you
+try to move something out of the match. So this could be something like, you
+match on an option or you match on a result. And when you pattern match on `Ok`,
+you try to take the value, move the value out of the `Ok` or `Err` variant, then
+Rust is going to move that value. If you had a guard that tried to use that
+value by reference—
+
+__Ben__: Hold up, what's a guard?
+
+__Jon__: So a guard is when you say something like `match r`, where `r` is a
+`Result` and you say, `Ok(v) if v > 5`.
+
+__Ben__: So it's— you have `if` in your pattern.
+
+__Jon__: Exactly, yeah. And specifically, previously if you moved in the
+pattern, but your guard had a reference, then the compiler would yell at you,
+and say you're not allowed to do that. Even though it should be totally fine,
+right? Because you move the value— or, you check the guard before you move the
+value. And then you move the value if the guard matches. So it should be totally
+fine. Previously, the compiler didn't understand this. Now it does.
+
+__Ben__: I think the fact that you can use `if`s in patterns is kind of also one
+of those more obscure features for us. That's very useful if you know how to do
+it. So hopefully this gets more people doing it.
+
+__Jon__: Oh, yeah. Match guards are— they're real handy when you need them.
+Sometimes they can just really clean up some some, like, hairy pattern matching
+code.
+
+__Ben__: Yeah. Um, what's next? I see one about attributes being enabled on
+function parameters.
+
+__Jon__: Yeah. So this is a fun one where, this usually happens if you have a
+crate to that needs to be able to run on multiple different platforms or
+targets. It can also happen in other circumstances. Imagine you have a function
+that takes like a— you have a low level library that operates on, like,
+operating system specific things. And so you want the argument to have one type
+on Windows, and one type on Linux, and one type on macOS. Previously, what you
+had to do is you had to have one instance of the function for each of those sort
+of configs. And then you had to replicate the entire signature and body of the
+function, only making the changes you needed.
+
+__Ben__: And because it's kind of like, you know, a little bit of boilerplate,
+people would probably often use macros for this. So nowadays you don't need to
+do that. Whereas before, you would just put an attribute on the function, and
+then have multiple copies of the function, each with different attributes for
+config'ing on and off. Now you can actually just have, like, whatever parameters
+need to be config'ed on and off, and just have the one function, which should
```

Comment:
I'm not sure how _config'ing_ and _config'ed_ should be spelled/punctuated. This seems least-unnatural to me, but feel free to weigh in if you have a preference.

Reply:
This seems okay to me :+1: